### ISSUE TITLE
Plugin chokes on typescript imports?

### ISSUE THREAD

**Issue by vedantroy** [dalcib/esbuild-plugin-cache#1]
I'm trying to use the `fs` module from Deno stdlib, and the plugin spits out the following errors:

```
 > deno-cache:https://x.nest.land/swc@0.0.6/mod.ts:17:28: error: Expected ")" but found ":"
    17 │ export function parse(source: string, opts: ParseOptions): Program {
       ╵                             ^

 > deno-cache:https://deno.land/std@0.95.0/fs/exists.ts:5:37: error: Expected ")" but found ":"
    5 │ export async function exists(filePath: string): Promise<boolean> {
      ╵                                      ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
    14 │ export async function ensureFile(filePath: string) {
       ╵                                          ^

 > deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
    11 │ export async function emptyDir(dir: string) {
       ╵                                   ^

 > deno-cache:https://deno.land/std@0.95.0/fs/expand_glob.ts:21:7: error: Unexpected "interface"
    21 │ export interface ExpandGlobOptions extends Omit<GlobOptions, "os"> {
       ╵        ~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/walk.ts:8:41: error: Expected ")" but found ":"
    8 │ export function _createWalkEntrySync(path: string): WalkEntry {
      ╵                                          ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
    9 │ export async function ensureDir(dir: string) {
      ╵                                    ^

 > deno-cache:https://deno.land/std@0.95.0/fs/eol.ts:4:7: error: Unexpected "enum"
    4 │ export enum EOL {
      ╵        ~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
    14 │ export async function ensureLink(src: string, dest: string) {
       ╵                                     ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_symlink.ts:15:39: error: Expected ")" but found ":"
    15 │ export async function ensureSymlink(src: string, dest: string) {
       ╵                                        ^

 > deno-cache:https://deno.land/std@0.95.0/fs/move.ts:5:10: error: Expected ";" but found "MoveOptions"
    5 │ interface MoveOptions {
      ╵           ~~~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
    8 │ export interface CopyOptions {
      ╵        ~~~~~~~~~

error: Uncaught (in promise) Error: Build failed with 12 errors:
deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
...
    at failureErrorWithLog (https://deno.land/x/esbuild@v0.11.17/mod.js:1398:15)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1080:28
    at runOnEndCallbacks (https://deno.land/x/esbuild@v0.11.17/mod.js:876:47)
    at buildResponseToResult (https://deno.land/x/esbuild@v0.11.17/mod.js:1078:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1185:14
    at https://deno.land/x/esbuild@v0.11.17/mod.js:567:9
    at handleIncomingPacket (https://deno.land/x/esbuild@v0.11.17/mod.js:664:9)
    at readFromStdout (https://deno.land/x/esbuild@v0.11.17/mod.js:534:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1663:11
?1 tanjiro-js % deno run --allow-read --allow-write --allow-env --allow-net --allow-run esbuild.ts
 > deno-cache:https://x.nest.land/swc@0.0.6/mod.ts:17:28: error: Expected ")" but found ":"
    17 │ export function parse(source: string, opts: ParseOptions): Program {
       ╵                             ^

 > deno-cache:https://deno.land/std@0.95.0/fs/move.ts:5:10: error: Expected ";" but found "MoveOptions"
    5 │ interface MoveOptions {
      ╵           ~~~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/exists.ts:5:37: error: Expected ")" but found ":"
    5 │ export async function exists(filePath: string): Promise<boolean> {
      ╵                                      ^

 > deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
    11 │ export async function emptyDir(dir: string) {
       ╵                                   ^

 > deno-cache:https://deno.land/std@0.95.0/fs/eol.ts:4:7: error: Unexpected "enum"
    4 │ export enum EOL {
      ╵        ~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/expand_glob.ts:21:7: error: Unexpected "interface"
    21 │ export interface ExpandGlobOptions extends Omit<GlobOptions, "os"> {
       ╵        ~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
    8 │ export interface CopyOptions {
      ╵        ~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_symlink.ts:15:39: error: Expected ")" but found ":"
    15 │ export async function ensureSymlink(src: string, dest: string) {
       ╵                                        ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
    9 │ export async function ensureDir(dir: string) {
      ╵                                    ^

 > deno-cache:https://deno.land/std@0.95.0/fs/walk.ts:8:41: error: Expected ")" but found ":"
    8 │ export function _createWalkEntrySync(path: string): WalkEntry {
      ╵                                          ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
    14 │ export async function ensureLink(src: string, dest: string) {
       ╵                                     ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
    14 │ export async function ensureFile(filePath: string) {
       ╵                                          ^

error: Uncaught (in promise) Error: Build failed with 12 errors:
deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
...
    at failureErrorWithLog (https://deno.land/x/esbuild@v0.11.17/mod.js:1398:15)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1080:28
    at runOnEndCallbacks (https://deno.land/x/esbuild@v0.11.17/mod.js:876:47)
    at buildResponseToResult (https://deno.land/x/esbuild@v0.11.17/mod.js:1078:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1185:14
    at https://deno.land/x/esbuild@v0.11.17/mod.js:567:9
    at handleIncomingPacket (https://deno.land/x/esbuild@v0.11.17/mod.js:664:9)
    at readFromStdout (https://deno.land/x/esbuild@v0.11.17/mod.js:534:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1663:11
?1 tanjiro-js % 
?1 tanjiro-js % 
?1 tanjiro-js % deno run --allow-read --allow-write --allow-env --allow-net --allow-run esbuild.ts
 > deno-cache:https://x.nest.land/swc@0.0.6/mod.ts:17:28: error: Expected ")" but found ":"
    17 │ export function parse(source: string, opts: ParseOptions): Program {
       ╵                             ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
    9 │ export async function ensureDir(dir: string) {
      ╵                                    ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_symlink.ts:15:39: error: Expected ")" but found ":"
    15 │ export async function ensureSymlink(src: string, dest: string) {
       ╵                                        ^

 > deno-cache:https://deno.land/std@0.95.0/fs/expand_glob.ts:21:7: error: Unexpected "interface"
    21 │ export interface ExpandGlobOptions extends Omit<GlobOptions, "os"> {
       ╵        ~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/move.ts:5:10: error: Expected ";" but found "MoveOptions"
    5 │ interface MoveOptions {
      ╵           ~~~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/eol.ts:4:7: error: Unexpected "enum"
    4 │ export enum EOL {
      ╵        ~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
    14 │ export async function ensureLink(src: string, dest: string) {
       ╵                                     ^

 > deno-cache:https://deno.land/std@0.95.0/fs/exists.ts:5:37: error: Expected ")" but found ":"
    5 │ export async function exists(filePath: string): Promise<boolean> {
      ╵                                      ^

 > deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
    11 │ export async function emptyDir(dir: string) {
       ╵                                   ^

 > deno-cache:https://deno.land/std@0.95.0/fs/walk.ts:8:41: error: Expected ")" but found ":"
    8 │ export function _createWalkEntrySync(path: string): WalkEntry {
      ╵                                          ^

 > deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
    8 │ export interface CopyOptions {
      ╵        ~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
    14 │ export async function ensureFile(filePath: string) {
       ╵                                          ^

error: Uncaught (in promise) Error: Build failed with 12 errors:
deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
...
    at failureErrorWithLog (https://deno.land/x/esbuild@v0.11.17/mod.js:1398:15)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1080:28
    at runOnEndCallbacks (https://deno.land/x/esbuild@v0.11.17/mod.js:876:47)
    at buildResponseToResult (https://deno.land/x/esbuild@v0.11.17/mod.js:1078:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1185:14
    at https://deno.land/x/esbuild@v0.11.17/mod.js:567:9
    at handleIncomingPacket (https://deno.land/x/esbuild@v0.11.17/mod.js:664:9)
    at readFromStdout (https://deno.land/x/esbuild@v0.11.17/mod.js:534:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1663:11
?1 tanjiro-js % deno run --allow-read --allow-write --allow-env --allow-net --allow-run esbuild.ts
 > deno-cache:https://x.nest.land/swc@0.0.6/mod.ts:17:28: error: Expected ")" but found ":"
    17 │ export function parse(source: string, opts: ParseOptions): Program {
       ╵                             ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_symlink.ts:15:39: error: Expected ")" but found ":"
    15 │ export async function ensureSymlink(src: string, dest: string) {
       ╵                                        ^

 > deno-cache:https://deno.land/std@0.95.0/fs/exists.ts:5:37: error: Expected ")" but found ":"
    5 │ export async function exists(filePath: string): Promise<boolean> {
      ╵                                      ^

 > deno-cache:https://deno.land/std@0.95.0/fs/eol.ts:4:7: error: Unexpected "enum"
    4 │ export enum EOL {
      ╵        ~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
    9 │ export async function ensureDir(dir: string) {
      ╵                                    ^

 > deno-cache:https://deno.land/std@0.95.0/fs/expand_glob.ts:21:7: error: Unexpected "interface"
    21 │ export interface ExpandGlobOptions extends Omit<GlobOptions, "os"> {
       ╵        ~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
    14 │ export async function ensureLink(src: string, dest: string) {
       ╵                                     ^

 > deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
    8 │ export interface CopyOptions {
      ╵        ~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
    11 │ export async function emptyDir(dir: string) {
       ╵                                   ^

 > deno-cache:https://deno.land/std@0.95.0/fs/move.ts:5:10: error: Expected ";" but found "MoveOptions"
    5 │ interface MoveOptions {
      ╵           ~~~~~~~~~~~

 > deno-cache:https://deno.land/std@0.95.0/fs/walk.ts:8:41: error: Expected ")" but found ":"
    8 │ export function _createWalkEntrySync(path: string): WalkEntry {
      ╵                                          ^

 > deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
    14 │ export async function ensureFile(filePath: string) {
       ╵                                          ^

error: Uncaught (in promise) Error: Build failed with 12 errors:
deno-cache:https://deno.land/std@0.95.0/fs/copy.ts:8:7: error: Unexpected "interface"
deno-cache:https://deno.land/std@0.95.0/fs/empty_dir.ts:11:34: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_dir.ts:9:35: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_file.ts:14:41: error: Expected ")" but found ":"
deno-cache:https://deno.land/std@0.95.0/fs/ensure_link.ts:14:36: error: Expected ")" but found ":"
...
    at failureErrorWithLog (https://deno.land/x/esbuild@v0.11.17/mod.js:1398:15)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1080:28
    at runOnEndCallbacks (https://deno.land/x/esbuild@v0.11.17/mod.js:876:47)
    at buildResponseToResult (https://deno.land/x/esbuild@v0.11.17/mod.js:1078:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1185:14
    at https://deno.land/x/esbuild@v0.11.17/mod.js:567:9
    at handleIncomingPacket (https://deno.land/x/esbuild@v0.11.17/mod.js:664:9)
    at readFromStdout (https://deno.land/x/esbuild@v0.11.17/mod.js:534:7)
    at https://deno.land/x/esbuild@v0.11.17/mod.js:1663:11
```

It seems like it doesn't expect Typescript syntax in remote imports?

**Comment by dalcib** [comment]
Nice catch. Thank you.
It is fixed in 5eb28d3257333605579677a2e3d8ef8ff70111fa
The new version is 0.2.6 
Please, let me know if it worked for you.

**Comment by vedantroy** [comment]
This worked! Thanks.
